LEGO1/lego/legoomni/include/act1state.h

```diff
@@ -25,29 +35,88 @@ class Act1State : public LegoState {
 	MxBool SetFlag() override;                          // vtable+0x18
 	MxResult VTable0x1c(LegoFile* p_legoFile) override; // vtable+0x1c
 
-	inline void SetUnknown18(MxU32 p_unk0x18) { m_unk0x18 = p_unk0x18; }
-	inline MxU32 GetUnknown18() { return m_unk0x18; }
-	inline MxU32 GetUnknown1c() { return m_unk0x1c; }
-	inline MxS16 GetUnknown21() { return m_unk0x21; }
+	inline void SetUnknown18(MxU32 p_unk0x18) { m_unk0x018 = p_unk0x18; }
+	inline MxU32 GetUnknown18() { return m_unk0x018; }
+	inline MxU32 GetUnknown1c() { return m_unk0x01c; }
+	inline MxS16 GetUnknown21() { return m_unk0x021; }
 
-	inline void SetUnknown1c(MxU32 p_unk0x1c) { m_unk0x1c = p_unk0x1c; }
-	inline void SetUnknown21(MxS16 p_unk0x21) { m_unk0x21 = p_unk0x21; }
+	inline void SetUnknown1c(MxU32 p_unk0x1c) { m_unk0x01c = p_unk0x1c; }
+	inline void SetUnknown21(MxS16 p_unk0x21) { m_unk0x021 = p_unk0x21; }
 
 	void FUN_10034d00();
 
 	// SYNTHETIC: LEGO1 0x10033960
 	// Act1State::`scalar deleting destructor'
 
+	// SIZE 0x4c
+	class NamedPlane {
+	public:
+		// FUNCTION: LEGO1 0x10033800
+		NamedPlane() {}
+
+		inline void SetName(const char* p_name) { m_name = p_name; }
+		inline const MxString* GetName() const { return &m_name; }
+		MxS32 Serialize(LegoFile* p_file);
+
+	private:
+		MxString m_name;         // 0x00
+		Mx3DPointFloat m_point1; // 0x10
+		Mx3DPointFloat m_point2; // 0x24
+		Mx3DPointFloat m_point3; // 0x38
+	};
+
+	// SIZE 0x14
+	class NamedTexture {
+	public:
+		~NamedTexture() { delete m_texture; }
+		// FUNCTION: LEGO1 0x1003f920
+		const MxString* GetName() const { return &m_name; }
+		LegoTexture* GetTexture() { return m_texture; }
+
+	private:
+		MxString m_name;        // 0x00
+		LegoTexture* m_texture; // 0x4
+	};
+
 protected:
-	undefined m_unk0x08[0x10]; // 0x08
-	MxU32 m_unk0x18;           // 0x18
-	undefined2 m_unk0x1c;      // 0x1c
-	undefined m_unk0x1e;       // 0x1e
-	undefined m_unk0x1f;       // 0x1f
-	undefined m_unk0x20;       // 0x20
-	MxBool m_unk0x21;          // 0x21
-	undefined m_unk0x22;       // 0x22
-							   // TODO
+	MxS32* m_unk0x008; // 0x008
+	// FIXME: count for m_unk0x008
+	MxS16 m_unk0x00c;         // 0x00c
+	undefined2 m_unk0x00e;    // 0x00e
+	undefined2 m_unk0x010;    // 0x010
+	undefined m_unk0x012;     // 0x012
+	MxS32 m_unk0x014;         // 0x014
+	MxU32 m_unk0x018;         // 0x018
+	MxU16 m_unk0x01c;         // 0x01c
+	undefined m_unk0x01e;     // 0x01e
+	undefined m_unk0x01f;     // 0x01f
+	undefined m_unk0x020;     // 0x020
+	undefined m_unk0x021;     // 0x021
+	undefined m_unk0x022;     // 0x022
+	undefined m_unk0x023;     // 0x023
+	NamedPlane m_unk0x024;    // 0x024
+	NamedPlane m_unk0x070;    // 0x070
+	NamedPlane m_unk0x0bc;    // 0x0bc
+	NamedPlane m_unk0x108;    // 0x108
+	NamedTexture* m_unk0x154; // 0x154
+	NamedTexture* m_unk0x158; // 0x158
+	NamedTexture* m_unk0x15c; // 0x15c
+	MxCore* m_unk0x160;       // 0x160
```

Comment:
This should obvious be a subclass of `MxCore`, but I guess `MxCore` will do for now?